src/modules/Embed/Embed.js

```diff
@@ -0,0 +1,183 @@
+import cx from 'classnames'
+import React, { PropTypes } from 'react'
+
+import {
+  AutoControlledComponent as Component,
+  customPropTypes,
+  getElementType,
+  getUnhandledProps,
+  META,
+  useKeyOnly,
+} from '../../lib'
+import Icon from '../../elements/Icon'
+import EmbedEmbed from './EmbedEmbed'
+
+const _meta = {
+  name: 'Embed',
+  type: META.TYPES.MODULE,
+  props: {
+    aspectRatio: ['4:3', '16:9', '21:9'],
+    source: ['youtube', 'vimeo'],
+  },
+}
+
+/**
+ * An embed displays content from other websites like YouTube videos or Google Maps.
+ */
+export default class Embed extends Component {
+  static autoControlledProps = [
+    'active',
+  ]
+
+  static defaultProps = {
+    icon: 'video play',
+  }
+
+  static _meta = _meta
+
+  static propTypes = {
+    /** An element type to render as (string or function). */
+    as: customPropTypes.as,
+
+    /** An embed can be active. */
+    active: PropTypes.bool,
+
+    /** An embed can specify an alternative aspect ratio. */
+    aspectRatio: PropTypes.oneOf(_meta.props.aspectRatio),
+
+    /** Primary content. */
+    children: PropTypes.node,
+
+    /** Additional classes. */
+    className: PropTypes.string,
+
+    /** Initial value of active. */
+    defaultActive: PropTypes.bool,
+
+    /** Specifies an icon to use with placeholder content. */
+    icon: customPropTypes.itemShorthand,
+
+    /** Сalled with event on Embed click with (event, props). */
+    onClick: PropTypes.func,
+
+    /** A placeholder image for embed. */
+    placeholder: PropTypes.string,
+
+    /** Specifies a source to use. */
+    source: customPropTypes.every([
+      customPropTypes.disallow(['sourceUrl']),
+      PropTypes.oneOf(_meta.props.source),
+    ]),
+
+    /** Specifies an id for source. */
+    sourceId: customPropTypes.every([
+      customPropTypes.demand(['source']),
+      PropTypes.string,
+    ]),
+
+    /** Settings to configure video behavior. */
+    sourceSettings: customPropTypes.every([
+      customPropTypes.demand(['source']),
+      PropTypes.shape({
```

Comment:
I think we should flatten these settings as top level props, instead of a nested object. They are applicable only to embedded videos, but I don't think that warrants nesting these props.  Let me know if you disagree and why.